MetaMask/metamask-extension
# restore pending approvals

## Explanation



One of the latest refactors to TokensController introduced an unplanned regression. This PR aims to fix it and includes E2E to verify that is the case.

In previous versions of MetaMask, a wallet_watchAsset request would persist if MetaMask is restarted.

## Manual Testing Steps

- Go to Test Dapp
- Move to section EIP 747 to trigger a wallet_watchAsset rpc call
- Instead of approving or declining the request, restart Metamask
- The wallet_watchAsset request should persist after restart

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
